### PR TITLE
Assigned to you tab for work allocation

### PR DESCRIPTION
**Pull request by ksp37-dluhc**
### Change description
This PR introduces the 'assigned to you' tab for users to view applications that have been assigned to them.

- [x] Unit tests and other appropriate tests added or updated
- [x] README and other documentation has been updated / added (if needed)
- [x] Commit messages are meaningful and follow good commit message guidelines (e.g. "FS-XXXX: Add margin to nav items preventing overlapping of logo")

### How to test
Currently an application will have to be manually assigned to a user (as the relevant UI pages have not been implemented). Once done, the user should be able to see the application on the 'assigned to you' tab. 

### Screenshots of UI changes (if applicable)

Screenshot 2024-07-11 at 15 19 35

Screenshot 2024-07-11 at 15 19 25

### CODE REVIEW

**Comment by Adam-W1** [comment]
Look great, nice work 👍 

only query is that the 'assigned to' column should have the user the one viewing the tab) in, I belive this is possible with the current API?

**Comment by ksp37-dluhc** [comment]
> Look great, nice work 👍
> 
> only query is that the 'assigned to' column should have the user the one viewing the tab) in, I belive this is possible with the current API?

Oh good catch! I'll fix that.

File: app/blueprints/assessments/routes.py [hunk 1]
```diff
@@ -267,10 +273,15 @@ def fund_dashboard(fund_short_name: str, round_short_name: str):
     future_tag_types = thread_executor.submit(get_tag_types)
 
     all_applications_metadata = future_all_applications_metadata.result()
+    applications_assigned_to_user = future_applications_assigned_to_user.result()
     application_overviews = future_application_overviews.result()
     active_fund_round_tags = future_active_fund_round_tags.result()
     tag_types = future_tag_types.result()
 
+    app_ids_assigned_to_user = set(
```
Comment (Adam-W1) [inline app/blueprints/assessments/routes.py:281]:
```suggestion
    app_ids_assigned_to_user = set(
        application["application_id"] for application in applications_assigned_to_user
    ) if applications_assigned_to_user else set()
```

Comment (ksp37-dluhc) [reply app/blueprints/assessments/routes.py]: Not using this code anywhere, so I'll mark it as resolved.

File: app/blueprints/assessments/helpers.py [hunk 2]
```diff
@@ -98,6 +98,29 @@ def set_application_status_in_overview(application_overviews):
     return application_overviews
 
 
+def set_assigned_info_in_overview(application_overviews, users_for_fund):
+    users_for_fund_dict = {user["account_id"]: user for user in users_for_fund}
```
Comment (Adam-W1) [inline app/blueprints/assessments/helpers.py:102]:
```suggestion
users_for_fund_dict = {user["account_id"]: user for user in users_for_fund} if users_for_fund else {}
```

Comment (Adam-W1) [reply app/blueprints/assessments/helpers.py:102]: ran into an error page locally when there are no users in the account store for the round (might not be a common scenario but this avoids this error page)

File: tests/test_routes.py [hunk 3]
```diff
@@ -216,6 +224,24 @@ def test_route_fund_dashboard(
...
         assert all(label in all_filter_labels for label in expected_filter_labels)
 
+        # Check the 'assigned to you' tab
+        assigned_to_you_table = (
+            soup.find("div", id="assigned-to-you")
+            .find("table", id="application_overviews_table")
+            .find("tbody")
+        )
+
+        expected_values = [
```
Comment (Adam-W1) [inline tests/test_routes.py:234]: could probs add the user data in here i think?

File: app/blueprints/assessments/helpers.py (hunk 2 above)
Comment (ksp37-dluhc) [reply app/blueprints/assessments/helpers.py:102]: Oh yep good shout, will update.

File: tests/test_routes.py (hunk 3 above)
Comment (ksp37-dluhc) [reply tests/test_routes.py:234]: Ah yes, missed that!

**Review by Adam-W1 — APPROVED** [review]
LGTM 👍

**Pull request merged**